### ISSUE TITLE
crash at startup due to undefined symbol "_ZN3nix16downloadSettingsE"

### ISSUE THREAD

**Issue by marius851000** [Mic92/nixpkgs-review#124]
how to reproduce:
1. clone the nixpkgs repository
2. ``cd nixpkgs``
3. ``git checkout fe7bab33d75cece4750fd1a51406a6e8993807ce``
4. ``nix-build . -A nixpkgs-review``
5. ``./result/bin/nixpkgs-review pr 95739``

It should crash here. If not, maybe those configuration may be relevant in the configuration.nix of nixos:
```
nix.package = pkgs.nixUnstable;
nix.extraOptions = ''
    experimental-features = nix-command flakes
'';
```

and I am nixpkgs revision 6a0fd33b4c15d7e0e0b0cdad5ef280eba32ccdcc

here is the log when I run the build command:
```
$ git -c fetch.prune=false fetch --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0 pull/95739/head:refs/nixpkgs-review/1
$ git worktree add /home/marius/.cache/nixpkgs-review/pr-95739-14/nixpkgs fe7bab33d75cece4750fd1a51406a6e8993807ce
Préparation de l'arbre de travail (HEAD détachée fe7bab33d75)
Mise à jour des fichiers: 100% (22453/22453), fait.
HEAD est maintenant à fe7bab33d75 Merge pull request #95553 from zowoq/rename-maintainers
$ git merge --no-commit 147e24dcd6988118c4c548c6ff5b0e1b62463037
La fusion automatique a réussi ; stoppée avant la validation comme demandé
nix: symbol lookup error: nix: undefined symbol: _ZN3nix16downloadSettingsE
https://github.com/NixOS/nixpkgs/pull/95739 failed to build
$ git worktree prune
```

**Comment by Mic92** [comment]
Sorry cannot reproduce with the commit. Do you have something weird in your environment?

Does it crash with?

````console
[nix-shell:~/git/nixpkgs]$ env - ./result/bin/nixpkgs-review pr 95739
```

Looks like nix is loading the wrong libraries or something.
Also do:

```console
$ LD_DEBUG=libs ./result/bin/nixpkgs-review pr 95739
```

**Comment by marius851000** [comment]
running ``env - ./result/bin/nixpkgs-review pr 95739`` sort of fixed the problem (it get stuck later without ever progressing nor using the CPU to compile), But I found that if I just unset ``LD_LIBRARY_PATH``, it seem to work. (also, I've switched to the pr 95790 due to it having less to compile).

I have attached the log when running ``LD_DEBUG=libs ./result/bin/nixpkgs-review pr 95790``:

[nixpkgs-review-ld-debug.log](https://github.com/Mic92/nixpkgs-review/files/5095367/nixpkgs-review-ld-debug.log)

Thank for your time.
